devicetypes/smartthings/zwave-mouse-trap.src/zwave-mouse-trap.groovy

```diff
@@ -0,0 +1,258 @@
+		result = createEvent(name: "pestControl", value: value, descriptionText: description)
+	} else if (cmd.notificationType == 0x13) {
+		//notificationType == 0x13 (Pest Control)
+		switch (cmd.event) {
+			case 0x00:
+				value = "idle"
+				description = "Trap cleared"
+				break
+			case 0x02:
+				value = "trapArmed"
+				description = "Trap armed"
+				break
+			case 0x04:
+				value = "trapRearmRequired"
+				description = "Trap re-arm required"
+				break
+			case 0x06:
+				value = "pestDetected"
+				description = "Pest detected"
+				break
+			case 0x08:
+				value = "pestExterminated"
+				description = "Pest exterminated"
+				break
+			default:
+				break
+		}
+		result = createEvent(name: "pestControl", value: value, descriptionText: description)
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd) {
+	log.debug "WakeUpNotification ${cmd}"
+	def event = createEvent(descriptionText: "${device.displayName} woke up", isStateChange: false)
+	def cmds = []
+	if (!state.MSR) {
+		cmds << zwave.manufacturerSpecificV2.manufacturerSpecificGet().format()
+		cmds << "delay 1200"
+	}
+	if (device.currentValue("pestControl") == null) { // In case our initial request didn't make it
+		cmds << zwave.notificationV3.notificationGet(notificationType: 0x13).format()
+	}
+	if (!state.lastbat || now() - state.lastbat > (24 * 60 * 60 + 6 * 60) * 1000 /*milliseconds*/) {
+		cmds << zwave.batteryV1.batteryGet().format()
+	} else {
+		// If we check the battery state we will send NoMoreInfo in the handler for BatteryReport so that we definitely get the report
+		cmds << zwave.wakeUpV1.wakeUpNoMoreInformation().format()
+	}
+	[event, response(cmds)]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def map = [name: "battery", unit: "%"]
+	if (cmd.batteryLevel == 0xFF) {
+		map.value = 1
+		map.descriptionText = "${device.displayName} has a low battery"
+		map.isStateChange = true
+	} else {
+		log.debug "Battery report: $cmd"
+		map.value = cmd.batteryLevel
+	}
+	state.lastbat = now()
+	[createEvent(map), response(zwave.wakeUpV1.wakeUpNoMoreInformation())]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+	def result = []
+
+	def msr = String.format("%04X-%04X-%04X", cmd.manufacturerId, cmd.productTypeId, cmd.productId)
+	log.debug "msr: $msr"
+	updateDataValue("MSR", msr)
+
+	result << createEvent(descriptionText: "$device.displayName MSR: $msr", isStateChange: false)
+
+	if (!device.currentState("battery")) {
+		result << response(zwave.batteryV1.batteryGet())
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	createEvent(descriptionText: "$device.displayName: $cmd", displayed: true)
+}
+
+def configure() {
+	log.debug "config"
+	response(getConfigurationCommands())
+}
+
+def getConfigurationCommands() {
+	log.debug "getConfigurationCommands"
+	def cmds = []
+	cmds << zwave.batteryV1.batteryGet().format()
```

Comment:
I'd remove this batteryGet from this method and just move it to installed by itself to prevent double-getting when you're attempting to configure again on wakeup.